pkg/minikube/download/preload.go

```diff
@@ -99,6 +92,36 @@ func remoteTarballURL(k8sVersion, containerRuntime string) string {
 	return fmt.Sprintf("https://storage.googleapis.com/%s/%s", PreloadBucket, TarballName(k8sVersion, containerRuntime))
 }
 
+func setPreloadState(k8sVersion, containerRuntime string, value bool) {
+	cRuntimes, ok := preloadStates[k8sVersion]
+	if !ok {
+		cRuntimes = make(map[string]bool)
+		preloadStates[k8sVersion] = cRuntimes
+	}
+	cRuntimes[containerRuntime] = value
+}
+
+var checkRemotePreloadExists = func(k8sVersion, containerRuntime string) bool {
+	url := remoteTarballURL(k8sVersion, containerRuntime)
+	resp, err := http.Head(url)
+	if err != nil {
+		klog.Warningf("%s fetch error: %v", url, err)
+		setPreloadState(k8sVersion, containerRuntime, false)
+		return false
+	}
+
+	// note: err won't be set if it's a 404
+	if resp.StatusCode != 200 {
+		klog.Warningf("%s status code: %d", url, resp.StatusCode)
+		setPreloadState(k8sVersion, containerRuntime, false)
+		return false
+	}
+
+	klog.Infof("Found remote preload: %s", url)
+	setPreloadState(k8sVersion, containerRuntime, true)
```

Comment:
Move it out of here.  Updating internal cache has nothing to do with checking remote state